src/System.Runtime.InteropServices/src/System/Runtime/InteropServices/RuntimeEnvironment.cs

```diff
@@ -23,7 +23,10 @@ public static bool FromGlobalAccessCache(System.Reflection.Assembly a)
         }
         public static string GetRuntimeDirectory()
         {
-            return Path.GetDirectoryName(typeof(object).Assembly.Location) + Path.DirectorySeparatorChar;
```

Comment:
It's probably looking for machine.cfg. You know, the one that exists only in memory and as a concept. This is why this assembly was raising my suspicions.